go-controller/pkg/node/gateway_iptables.go

```diff
@@ -532,7 +538,8 @@ func recreateIPTRules(table, chain string, keepIPTRules []nodeipt.Rule) error {
 			errors = append(errors, fmt.Errorf("error clearing Chain: %s in Table: %s, err: %v", chain, table, err))
 		}
 	}
-	if err = insertIptRules(keepIPTRules); err != nil {
+	filter := map[string]map[string]bool{table: {chain: false}}
```

Comment:
No I just used bool instead of empty struct. If you prefer the latter I can change it, or I can add a comment saying the value doesnt matter in the function.

Reply:
comment is fine,

Reply:
NIT, feel free to ignore it: I thought that the point of implementing a set as `map[string]bool` is to set the value to true for all keys we're adding, so that we can check for existence with `if filter[r.Table][r.Chain] {...}`, but I see that when you index the filter you do `if _, ok := filter[r.Table][r.Chain]; !ok`, which is of course the safest way of doing it. Maybe `map[string]struct{}` is easier to understand in this case?